x/mongo/driver/topology/connection.go

```diff
@@ -417,6 +417,8 @@ func (c *connection) readWireMessage(ctx context.Context, dst []byte) ([]byte, e
 }
 
 func (c *connection) read(ctx context.Context, dst []byte) (bytesRead []byte, errMsg string, err error) {
+	// Clear dst in case it came from a memory pool.
+	dst = dst[:0]
```

Comment:
Is this always safe to do? There are some callers of `ReadWireMessage` outside of the driver that may expect it to always append to the end of the `dst` slice.